libr/anal/esil.c

```diff
@@ -544,6 +544,42 @@ R_API int r_anal_esil_reg_read(RAnalEsil *esil, const char *regname, ut64 *num,
 	return ret;
 }
 
+// sign extension operator for use in idiv, imul, movsx* 
+// and other instructions involving signed values
```

Comment:
Better add some comments like examples:
```
<bit-width of value>,<value>,~
"ae 8,0xfb,~"
```

Reply:
Actually, more detailed can also be added: Extend signed integer to the size of st64 on machine running this ESIL vm.
Since my opinion come from the view of P-code to ESIL translation, I will care about something like this while others may take this as granted. So whether to add is up to you.
And `~/` and `~%` do not need extra comments as long as them works just like normal `/` and `%`.

Reply:
Anyway, I hope this pr will be merged sooner.

Reply:
Yea having a signext operator will help with all different kinds of IR translation. I am not sure if the VEX2ESIL tool I have in my ESILSolve repo could be useful to the pcode stuff but there are probably similar problems involved